File: Ingineria sistemelor soft/aplicatia/temaISS/model/src/main/java/com/example/Entity.java
package com.example;

import javax.persistence.Column;
import javax.persistence.Id;
import java.io.Serializable;

public abstract class Entity<ID> implements Serializable {

    @Id
    @Column(name = "id")
    private ID id;

    public ID getId() {
        return id;
    }

    public void setId(ID id) {
        this.id = id;
    }
}

File: Project/tripModel/Entity.cs
using System;

namespace tripModel
{
    [Serializable]
    public class Entity<ID>
    {
        private ID id;
        
        public Entity(ID id)
        {
            this.id = id;
        }

        public ID Id
        {
            get { return id; }

            set { id = value; }
        }

    }
}